src/main/java/org/cactoos/list/ListIteratorJoined.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.list;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A few {@link ListIterator} joined together.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <T> Items type
+ * @since 0.49
+ */
+public final class ListIteratorJoined<T> implements ListIterator<T> {
+
+    /**
+     * ListIterators.
+     */
+    private final ListIterator<ListIterator<T>> listiters;
+
+    /**
+     * Current traversal list iterator.
+     */
+    private AtomicReference<ListIterator<T>> current;
+
+    /**
+     * Cursor.
+     */
+    private final AtomicInteger cursor;
+
+    /**
+     * Ctor.
+     * @param items Items to concatenate
+     */
+    @SafeVarargs
+    public ListIteratorJoined(final ListIterator<T>... items) {
+        this(new ListOf<>(items));
+    }
+
+    /**
+     * Ctor.
+     * @param item First item
+     * @param items ListIterator
+     */
+    @SuppressWarnings("unchecked")
+    public ListIteratorJoined(final T item, final ListIterator<T> items) {
+        this(new ListOf<>(new ListOf<>(item).listIterator(), items));
+    }
+
+    /**
+     * Ctor.
+     * @param items ListIterators
+     * @param item End item
+     */
+    @SuppressWarnings("unchecked")
+    public ListIteratorJoined(final ListIterator<T> items, final T item) {
+        this(new ListOf<>(items, new ListOf<>(item).listIterator()));
+    }
+
+    /**
+     * Ctor.
+     * @param items Items to concatenate
+     */
+    public ListIteratorJoined(final List<ListIterator<T>> items) {
+        this.listiters = items.listIterator();
```

Comment:
@victornoel I'm afraid that no. To make it works, we need to store ListIterator of each lists to be able to switch from one list to another. You could better understand by looking at `iterator.Joined`. Because `ListIteratorJoined` and `list.Joined` sould respect the same logic as `iterator.Joined` and ` iterable.Joined` . So, let's try your second suggestion.